arrays/arrays.c

```diff
@@ -175,12 +175,34 @@ void arr_remove(Array *arr, char *element)
 
   // Search for the first occurence of the element and remove it.
   // Don't forget to free its memory!
+  int i = 0;
+  while (i < arr->count)
+  {
+    if (arr->elements[i] == element)
+    {
+      arr->elements[i] = NULL;
```

Comment:
nice safeguard, but I think you may bee able to do without it. once you get it to work well, try and see if you can remove this line.